sound/soc/sof/sof-client.h

```diff
@@ -52,6 +56,27 @@ int sof_client_ipc_tx_message(struct sof_client_dev *cdev, u32 header, void *msg
 			      size_t msg_bytes, void *reply_data, size_t reply_bytes);
 
 struct dentry *sof_client_get_debugfs_root(struct sof_client_dev *cdev);
+struct device *sof_client_get_dma_dev(struct sof_client_dev *cdev);
```

Comment:
this could be under the `#ifdef` too?

Reply:
@lyakh this API will be needed for any client that needs to alloc memory pages for DMA and I didnt want to restrict it to just probes alone.

Reply:
probably worth adding a comment that this is generic enough to be used by other clients, and is required by probes to extract the information with DMA transfers.